[FIX] 14.0 cash rounding on invoices

PierrickBrun opened the pull request.

Commit:
[FIX] cash rounding on invoices

Commit:
[FIX] cash rounding only on cash methods

hparfr commented on pos_sale_order/models/sale_order.py:
```diff
@@ -206,6 +206,11 @@ def action_pos_order_invoice(self):
         res["session_id"] = self.session_id.id
+        if self.config_id.cash_rounding and (
+            not self.config_id.only_round_cash_method
+            or any(p.payment_method_id.is_cash_count for p in self.payment_ids)
```
it's not the good algorithm, we should take the one from odoo/point_of_sale : round only if the last of the payment_method_id is cash_count

hparfr committed:
pos_sale_order: fix cash rounding

if there is cash then bank, no rounding
should be applied

hparfr committed:
pos_sale_order: cash rounding implement better algo

when you pay first by gift card then you pay by cash
cash rounding should be taken in account.
